compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1031,17 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(sym: Symbol): Unit = {
+    for {
+      ann <- compilerInternalAnnotations
+      annSym <- sym.getAnnotation(ann)
+      if annSym.pos.isDefined
```

Comment:
In practice, the position is only defined if the annotation came from the file. I added it to filter out the `@WasPublicBeforeTyper` that were generated in `PreTyperComponent`.